### PR TITLE
revert(*): revert resource-group, storage-account, appservice-windows and keyvault changes

### PR DESCRIPTION
**Pull request by filipefigueredo**
# Description

The goal of this PR is to revert unintended changes in the  following modules:
- keyvault
- app-service-windows
- resource-group
- storage-account

# Motivation and Context

At the  PR #39, I had my code base outdated and merged old stuff from main branch that changed behaviors previously updated.

# How Has This Been Tested?

Please compare the reverted changes with the commit `952ead1`
# Screenshots

# Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [ ] My code has security implications.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: azure/storage_account/data.tf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+data "azurerm_resource_group" "rg" {
+  name = var.resource_group_name
+}
```
Comment (fargozhu) [inline azure/storage_account/data.tf:3]: [recommendation] I would move the data block to the main file instead, so we can easily follow the flow in one place only.

File: azure/storage_account/main.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
+resource "random_id" "storage_account_name" {
```
Comment (fargozhu) [inline azure/storage_account/main.tf:1]:
```suggestion
data "azurerm_resource_group" "rg" {
  name = var.resource_group_name
}

resource "random_id" "storage_account_name" {
```

File: azure/storage_account/main.tf [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+resource "azurerm_storage_account" "storage_account" {
+  name                      = random_id.storage_account_name.hex
+  resource_group_name       = data.azurerm_resource_group.rg.name
+  location                  = data.azurerm_resource_group.rg.location
+  account_kind              = var.kind
+  account_tier              = var.account_tier
+  account_replication_type  = var.replication_type
+  enable_https_traffic_only = var.enable_https_traffic_only
+  min_tls_version           = var.min_tls_version
```
Comment (fargozhu) [inline azure/storage_account/main.tf:26]:
```suggestion
  min_tls_version           = var.min_tls_version
  
  tags = data.azurerm_resource_group.rg.tags
```

**Review by fargozhu — APPROVED** [review]
All comments and suggests are nice to have, but I am still concerned with the storage account module and the fact that is getting the location value from the resource group data block.

**Review by fargozhu — CHANGES_REQUESTED** [review]
Most are comments, but the missing tags on the storage account needs to be added.

**Comment by filipefigueredo** [comment]
All the suggestions for storage account were implemented.

**Review by fargozhu — APPROVED** [review]

**Pull request merged**